### PR TITLE
Code style formatting using black + isort

### PR DESCRIPTION
**Pull request by kalaspuff**
* Added code style formatting using `black` and `isort`.
  * https://github.com/psf/black (The Uncompromising Code Formatter)
  * https://pycqa.github.io/isort (sorts imports alphabetically, and automatically separated into sections and by type)

* Added `flake8` to dev-dependencies (incl. `flake8-black` + `flake8-isort`) for future linting (and potential editor niceness / developer delightness). 
  * Note: Since `flake8` doesn't support PEP 518 (`pyproject.toml`) configuration yet, we'll have to deal with pesky `tox.ini` (or `.flake8` / `setup.cfg`) files for the time being.

### Rationale
My take is that it may be good to apply these kinds of tooling on projects from the get-go as soon as possible to build some kind of understanding of how to align devs work, even if they're coming from different schools. I think black has done a great job at creating as few options as possible so that devs won't have to end up in endless discussions regarding "single quotes vs double quotes", "2 spaces vs 4 spaces", "trailing comma at end of arrays or not", "how many newlines should there be before a new function outside of a class", etc. 

Using tools such as `black` and `isort` in new projects (I'm not full-on advocating to always reformat existing larger codebases though) should make us avoid potential discussions regarding what code styles people prefer. Also makes PR:s / code diffs cleaner.

Getting things like that in early is usually easier than later having to deal with politics around different code styles to choose later on, and also making a larger dent if having to deal with merge conflicts to a larger extent if code formatting ruling isn't applied very early on.

> “Any color you like. Black.”

### CODE REVIEW

**Review by pedropregueiro — COMMENTED** [review]
Is there anything worth adding to the README.md? I'm thinking, as a pycharm or vscode user, what should I do to avoid discrepancies in formatting on future PRs. 

Or maybe this can be part of a more structured `# Contributing` section in a later PR

File: pyproject.toml [hunk 1]
```diff
@@ -34,9 +34,26 @@ watchgod = "^0.7"
 web3 = "^5.26"
 
 [tool.poetry.dev-dependencies]
+black = { version = "*", allow-prereleases = true }
```
[the remark this answers is not in the document]
Comment (kalaspuff) [reply pyproject.toml:37]: All `black` versions yet released are actually prereleases (denominated by the `a` and `b` in their version tags) and therefore needs the `allow-prereleases` option to find available releases during version resolution. https://github.com/psf/black#note-this-is-a-beta-product

Screenshot 2022-01-17 at 22 36 04
Screenshot 2022-01-17 at 22 35 19

> Also, as a safety measure which slows down processing, Black will check that the reformatted code still produces a valid AST that is effectively equivalent to the original

**Comment by kalaspuff** [comment]
> Is there anything worth adding to the README.md? I'm thinking, as a pycharm or vscode user, what should I do to avoid discrepancies in formatting on future PRs.
> 
> Or maybe this can be part of a more structured `# Contributing` section in a later PR

I think it makes sense to add that kinds of contributor info to a `CONTRIBUTING.md`.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedropregueiro) [reply pyproject.toml:37]: good to know

**Pull request merged**